doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -436,34 +447,93 @@ is_midplane_reachable(self):
+* The dpu operational state transition from 'offline' to 'online' guarantees the reboot of a DPU.
+* The platform code will extract the DPU reboot cause from the NPU hardware itself even when the DPU is not reachable.
+* The DPU reboot cause will be mapped to one of the following existing reboot causes and returned back.
+    * REBOOT_CAUSE_POWER_LOSS = "Power Loss"
+    * REBOOT_CAUSE_THERMAL_OVERLOAD_CPU = "Thermal Overload: CPU"
+    * REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC = "Thermal Overload: ASIC"
+    * REBOOT_CAUSE_THERMAL_OVERLOAD_OTHER = "Thermal Overload: Other"
+    * REBOOT_CAUSE_INSUFFICIENT_FAN_SPEED = "Insufficient Fan Speed"
+    * REBOOT_CAUSE_WATCHDOG = "Watchdog"
+    * REBOOT_CAUSE_HARDWARE_OTHER = "Hardware - Other"
+    * REBOOT_CAUSE_HARDWARE_BIOS = "BIOS"
+    * REBOOT_CAUSE_HARDWARE_CPU = "CPU"
+    * REBOOT_CAUSE_HARDWARE_BUTTON = "Push button"
+    * REBOOT_CAUSE_HARDWARE_RESET_FROM_ASIC = "Reset from ASIC"
+    * REBOOT_CAUSE_NON_HARDWARE = "Non-Hardware"
+* When the platform wants to provide additional details to the reboot cause they can append it to the string with a ',' separation as shown below.
+```
+return REBOOT_CAUSE_NON_HARDWARE + ', ' + 'kernel panic'
+```
+* The switch reboot use case will follow the same sequence.
+* The pmon container restart should not affect this sequence as the states are persisted either in the DB or in the file system.
+
+#### Schema for REBOOT_CAUSE of DPUs on switch ChassisStateDB
+```
+  Key: "REBOOT_CAUSE|DPU0|2024_06_06_09_31_18"
+
+  "REBOOT_CAUSE|DPU0|2024_06_06_09_31_18": {
+    "value": {
+      "cause": "Software causes (Reboot)",
+      "comment": "User issued 'reboot' command [User: admin, Time: Thu Jun  6 09:46:43 AM UTC 2024]",
+      "device": "DPU0",
+      "time": "N/A",
+      "user": "N/A"
+    }
+  }
 
 ```
-2. Though the get_oper_status(self) can get the operational status of the DPU Modules, the current implementation only has limited capabilities.
+#### 2. DPU State
+Though the get_oper_status(self) can get the operational status of the DPU modules, the current implementation only has limited capabilities.
     * Can only state MODULE_STATUS_FAULT and can't show exactly where in the state progression the DPU failed. This is critical in fault isolation, DPU switchover decision, resiliency and recovery
     * Though this is platform implementation specific, in a multi vendor use case, there has to be a consistent way of storing and accessing the information.
-    * Store the state progression (dpu_midplane_link_state, dpu_control_plane_state, dpu_data_plane_state) on the host ChassisStateDB.
+    * Store the state progression (dpu_midplane_link_state, dpu_control_plane_state, dpu_data_plane_state) on the host ChassisStateDB using the push model specified in [section: 3.2.4 of SONiC Chassis Platform Management & Monitoring HLD](https://github.com/sonic-net/SONiC/blob/master/doc/pmon/pmon-chassis-design.md)
```

Comment:
We agreed that the `dpu_control_plane_state` and `dpu_data_plane_state` will be pushed by the DPU. But the `dpu_midplane_link_state` should be monitored and set by the NPU. Please update this.

Reply:
@oleksandrivantsiv Updated

Reply:
#closed